<launch>

<!-- these are the arguments you can pass this launch file, for example paused:=true -->
<arg name="paused" default="false"/>
<arg name="use_sim_time" default="true"/>
<arg name="extra_gazebo_args" default=""/>
<arg name="gui" default="true"/>
<arg name="headless" default="false"/>
<arg name="debug" default="false"/>
<arg name="physics" default="ode"/>
<arg name="verbose" default="false"/>
<arg name="world_name" default="worlds/empty.world" />
<!-- Note: the world_name is with respect to GAZEBO_RESOURCE_PATH environmental variable -->


<!-- arg name="fcu_url" default="serial:///dev/ttyUSB0:921600" -->
<arg name="fcu_url" default="udp://:14540@127.0.0.1:14557" />
<arg name="gcs_url" default="" />
<arg name="tgt_system" default="1" />
<arg name="tgt_component" default="1" />

<!-- set use_sim_time flag -->
<group if="$(arg use_sim_time)">
	<param name="/use_sim_time" value="true" />
</group>

<!-- set command arguments -->
<arg unless="$(arg paused)" name="command_arg1" value=""/>
<arg     if="$(arg paused)" name="command_arg1" value="-u"/>
<arg unless="$(arg headless)" name="command_arg2" value=""/>
<arg     if="$(arg headless)" name="command_arg2" value="-r"/>
<arg unless="$(arg verbose)" name="command_arg3" value=""/>
<arg     if="$(arg verbose)" name="command_arg3" value="--verbose"/>
<arg unless="$(arg debug)" name="script_type" value="gzserver"/>
<arg     if="$(arg debug)" name="script_type" value="debug"/>

<!-- start gazebo server-->
<node name="gazebo" pkg="gazebo_ros" type="$(arg script_type)" respawn="false" output="screen"
args="$(arg command_arg1) $(arg command_arg2) $(arg command_arg3) -e $(arg physics) $(arg extra_gazebo_args) $(arg world_name)" />

<!-- start gazebo client -->
<group if="$(arg gui)">
<node name="gazebo_gui" pkg="gazebo_ros" type="gzclient" respawn="false" output="screen"/>
</group>

<node name="mavros" pkg="mavros" type="mavros_node" output="screen">
	<param name="fcu_url" value="$(arg fcu_url)" />
	<param name="gcs_url" value="$(arg gcs_url)" />
	<param name="target_system_id" value="$(arg tgt_system)" />
	<param name="target_component_id" value="$(arg tgt_component)" />

	<!-- enable heartbeat send and reduce timeout -->
	<param name="conn_heartbeat" value="5.0" />
	<param name="conn_timeout" value="5.0" />
	<!-- automatically start mavlink on USB -->
	<param name="startup_px4_usb_quirk" value="true" />

</node>

<node name="fixedtool" pkg="inrol_fixedtool" type="torque_fixedtool_sitl_e" output="screen">

	<param name="gains/pos_P" value = "4.00"/>
	<param name="gains/pos_D" value = "1.0"/>
	<param name="gains/att_k_star" value = "0.1"/>

	<param name="tool/d1" value = "0.1"/>
	<param name="tool/d2" value = "0.0"/>
	<param name="tool/d3" value = "-0.5"/>
	<param name ="mass" value = "1.5" />
	<!-- ctrl_mode true: !-->
	<param name="ctrl_mode" value = "true" />
	<!-- estimation_mode true: onboard estimation false: Based on true values from gazebo-->
	<param name="estimation_mode" value = "false" />

	<remap from = "/fixedtool/actuator_control" to = "/mavros/actuator_control" />

</node>

</launch>
